server-spi/src/main/java/org/keycloak/models/OrganizationModel.java

```diff
@@ -26,4 +26,6 @@ public interface OrganizationModel {
     void setName(String name);
 
     String getName();
+
+    GroupModel getGroup();
```

Comment:
I agree with this sentiment, making the model contract not depend on the underlying data representation encapsulates the functionality well.

Reply:
Thinking on this, I can see the desire to do it this way so that you can enable some behavior like 

`getOrg -> getGroup -> use existing code to modify attributes` and you avoid having to repeat code that's already written in the application layer. 

I feel like the trade-off of repeating some behavior through a well-expressed contract is worth it in this case though. Keeping the data access details encapsulated to the data access layer and then freeing the model object from a dependency follows the level of flexibility that keycloak usually aims for, in my opinion.